Server crash with panic: runtime error: index out of range [4] with length 4

[2021-03-18 11:09:00.251] [DEBUG] [peer.go:116] => On subscriber ice candidate called for peer peer_id=ckmeao2k80003gr17lab5nmcr v=1
[2021-03-18 11:09:00.251] [DEBUG] [peer.go:116] => On subscriber ice candidate called for peer peer_id=ckmeao2k80003gr17lab5nmcr v=1
[2021-03-18 11:09:00.251] [DEBUG] [peer.go:116] => On subscriber ice candidate called for peer peer_id=ckmeao2k80003gr17lab5nmcr v=1
[2021-03-18 11:09:00.363] [INFO] [peer.go:184] => Peer got answer peer_id=ckmeao2k80003gr17lab5nmcr v=0
[2021-03-18 11:09:00.363] [DEBUG] [subscriber.go:56] => ice connection status state=2 v=1
[2021-03-18 11:09:00.370] [INFO] [peer.go:204] => Peer trickle peer_od=ckmeao2k80003gr17lab5nmcr v=0
[2021-03-18 11:09:01.407] [DEBUG] [subscriber.go:56] => ice connection status state=3 v=1
[2021-03-18 11:09:02.562] [INFO] [peer.go:160] => Peer got offer peer_id=ckmeao2k80003gr17lab5nmcr v=0
[2021-03-18 11:09:02.564] [INFO] [peer.go:171] => Peer send answer peer_id=ckmeao2k80003gr17lab5nmcr v=0
[2021-03-18 11:09:02.569] [INFO] [peer.go:204] => Peer trickle peer_od=ckmeao2k80003gr17lab5nmcr v=0
[2021-03-18 11:09:02.572] [INFO] [peer.go:204] => Peer trickle peer_od=ckmeao2k80003gr17lab5nmcr v=0
[2021-03-18 11:09:02.615] [DEBUG] [publisher.go:47] => Peer got remote track id mediaSSRC=3329196902 peer_id=ckmeao2k80003gr17lab5nmcr rid= stram_id=e1f7747b-f2c4-4183-b3b8-3c8a7d229991 track_id=eaec036c-2f8b-41f2-8cf1-da7b8384bf68 v=1
[2021-03-18 11:09:02.615] [INFO] [session.go:138] => Publishing track to peer peer_id=ckmeanrrc0002gr179yiy4nol v=0
[2021-03-18 11:09:02.616] [INFO] [session.go:138] => Publishing track to peer peer_id=ckmean8gw0001gr17tzk5nw75 v=0
[2021-03-18 11:09:02.866] [DEBUG] [peer.go:101] => Negotiation needed peer_id=ckmeanrrc0002gr179yiy4nol v=1
[2021-03-18 11:09:02.866] [DEBUG] [peer.go:101] => Negotiation needed peer_id=ckmean8gw0001gr17tzk5nw75 v=1
[2021-03-18 11:09:02.867] [INFO] [peer.go:110] => Send offer peer_id=ckmeanrrc0002gr179yiy4nol v=0
[2021-03-18 11:09:02.867] [INFO] [peer.go:110] => Send offer peer_id=ckmean8gw0001gr17tzk5nw75 v=0
[2021-03-18 11:09:02.876] [INFO] [peer.go:184] => Peer got answer peer_id=ckmean8gw0001gr17tzk5nw75 v=0
[2021-03-18 11:09:02.880] [INFO] [peer.go:184] => Peer got answer peer_id=ckmeanrrc0002gr179yiy4nol v=0
[2021-03-18 11:09:03.118] [DEBUG] [publisher.go:47] => Peer got remote track id mediaSSRC=3053033019 peer_id=ckmeao2k80003gr17lab5nmcr rid= stram_id=e1f7747b-f2c4-4183-b3b8-3c8a7d229991 track_id=fd47096c-2369-415a-a478-5c7dcbe9d70f v=1
[2021-03-18 11:09:03.120] [INFO] [session.go:138] => Publishing track to peer peer_id=ckmeanrrc0002gr179yiy4nol v=0
[2021-03-18 11:09:03.120] [INFO] [session.go:138] => Publishing track to peer peer_id=ckmean8gw0001gr17tzk5nw75 v=0
[2021-03-18 11:09:03.371] [DEBUG] [peer.go:101] => Negotiation needed peer_id=ckmeanrrc0002gr179yiy4nol v=1
[2021-03-18 11:09:03.371] [DEBUG] [peer.go:101] => Negotiation needed peer_id=ckmean8gw0001gr17tzk5nw75 v=1
[2021-03-18 11:09:03.371] [INFO] [peer.go:110] => Send offer peer_id=ckmeanrrc0002gr179yiy4nol v=0
[2021-03-18 11:09:03.372] [INFO] [peer.go:110] => Send offer peer_id=ckmean8gw0001gr17tzk5nw75 v=0
[2021-03-18 11:09:03.378] [INFO] [peer.go:184] => Peer got answer peer_id=ckmean8gw0001gr17tzk5nw75 v=0
[2021-03-18 11:09:03.379] [ERROR] [subscriber.go:192] => SetRemoteDescription error error="unable to start track, codec is not supported by remote"
[2021-03-18 11:09:03.381] [INFO] [peer.go:184] => Peer got answer peer_id=ckmeanrrc0002gr179yiy4nol v=0
[2021-03-18 11:09:03.381] [ERROR] [subscriber.go:192] => SetRemoteDescription error error="unable to start track, codec is not supported by remote"
panic: runtime error: index out of range [4] with length 4

goroutine 275 [running]:
github.com/pion/ion-sfu/pkg/buffer.(*VP8).Unmarshal(0xc000267b10, 0xc0007ab05e, 0x4, 0x13c5a, 0x0, 0x0)
	/home/chen_caixiong/go/src/github.com/pion/ion-sfu/pkg/buffer/helpers.go:101 +0x231
github.com/pion/ion-sfu/pkg/buffer.(*Buffer).calc(0xc0001a81a0, 0xc000c6b5c0, 0x10, 0x20, 0x166d5170762d534e)
	/home/chen_caixiong/go/src/github.com/pion/ion-sfu/pkg/buffer/buffer.go:332 +0x8fd
github.com/pion/ion-sfu/pkg/buffer.(*Buffer).Write(0xc0001a81a0, 0xc000c6b5c0, 0x10, 0x20, 0x0, 0x0, 0x0)
	/home/chen_caixiong/go/src/github.com/pion/ion-sfu/pkg/buffer/buffer.go:206 +0x144
github.com/pion/srtp/v2.(*ReadStreamSRTP).write(0xc000468a80, 0xc000c6b5c0, 0x10, 0x20, 0xc0002f8000, 0x20, 0x2000)
	/home/chen_caixiong/go/pkg/mod/github.com/pion/srtp/v2@v2.0.1/stream_srtp.go:64 +0x56
github.com/pion/srtp/v2.(*SessionSRTP).decrypt(0xc0003d6f20, 0xc0002f8000, 0x20, 0x2000, 0x20, 0x0)
	/home/chen_caixiong/go/pkg/mod/github.com/pion/srtp/v2@v2.0.1/session_srtp.go:165 +0x1b2
github.com/pion/srtp/v2.(*session).start.func1(0xc0003d6f20, 0xc000185940, 0xfbf420, 0xc0003d6f20)
	/home/chen_caixiong/go/pkg/mod/github.com/pion/srtp/v2@v2.0.1/session.go:141 +0x14c
created by github.com/pion/srtp/v2.(*session).start
	/home/chen_caixiong/go/pkg/mod/github.com/pion/srtp/v2@v2.0.1/session.go:120 +0x226



[2021-03-18 11:06:34.571] [DEBUG] [subscriber.go:56] => ice connection status state=3 v=1
[2021-03-18 11:06:34.588] [INFO] [peer.go:204] => Peer trickle peer_od=ckmeakyog0002cc171x91eu44 v=0
[2021-03-18 11:06:34.612] [INFO] [peer.go:184] => Peer got answer peer_id=ckmeakutk0000cc176qouyft0 v=0
[2021-03-18 11:06:34.613] [ERROR] [subscriber.go:192] => SetRemoteDescription error error="unable to start track, codec is not supported by remote"
[2021-03-18 11:06:34.637] [INFO] [peer.go:204] => Peer trickle peer_od=ckmeakyog0002cc171x91eu44 v=0
[2021-03-18 11:06:34.725] [INFO] [peer.go:184] => Peer got answer peer_id=ckmeakvlc0001cc17pekoluob v=0
panic: runtime error: index out of range [4] with length 4

goroutine 394 [running]:
github.com/pion/ion-sfu/pkg/buffer.(*VP8).Unmarshal(0xc0008ccb10, 0xc00095290e, 0x4, 0x263aa, 0x0, 0x0)
	/home/chen_caixiong/go/src/github.com/pion/ion-sfu/pkg/buffer/helpers.go:105 +0x1e0
github.com/pion/ion-sfu/pkg/buffer.(*Buffer).calc(0xc0002329c0, 0xc00039a5c0, 0x10, 0x20, 0x166d50eb6a900fde)
	/home/chen_caixiong/go/src/github.com/pion/ion-sfu/pkg/buffer/buffer.go:332 +0x8fd
github.com/pion/ion-sfu/pkg/buffer.(*Buffer).Write(0xc0002329c0, 0xc00039a5c0, 0x10, 0x20, 0x0, 0x0, 0x0)
	/home/chen_caixiong/go/src/github.com/pion/ion-sfu/pkg/buffer/buffer.go:206 +0x144
github.com/pion/srtp/v2.(*ReadStreamSRTP).write(0xc0004156c0, 0xc00039a5c0, 0x10, 0x20, 0xc0004b2000, 0x20, 0x2000)
	/home/chen_caixiong/go/pkg/mod/github.com/pion/srtp/v2@v2.0.1/stream_srtp.go:64 +0x56
github.com/pion/srtp/v2.(*SessionSRTP).decrypt(0xc0002628f0, 0xc0004b2000, 0x20, 0x2000, 0x20, 0x0)
	/home/chen_caixiong/go/pkg/mod/github.com/pion/srtp/v2@v2.0.1/session_srtp.go:165 +0x1b2
github.com/pion/srtp/v2.(*session).start.func1(0xc0002628f0, 0xc0001821f0, 0xfbf420, 0xc0002628f0)
	/home/chen_caixiong/go/pkg/mod/github.com/pion/srtp/v2@v2.0.1/session.go:141 +0x14c
created by github.com/pion/srtp/v2.(*session).start
	/home/chen_caixiong/go/pkg/mod/github.com/pion/srtp/v2@v2.0.1/session.go:120 +0x226
exit status 2

--- a/pkg/buffer/helpers.go
+++ b/pkg/buffer/helpers.go
@@ -98,6 +98,10 @@ func (p *VP8) Unmarshal(payload []byte) error {
 		if L {
 			idx++
 			p.TlzIdx = idx
+
+			if int(idx) >= payloadLen {
+				return errShortPacket
+			}
 			p.TL0PICIDX = payload[idx]
 		}
 		if p.TemporalSupported || K {

